src/nsls2api/cli/api.py

```diff
@@ -1,13 +1,185 @@
+import httpx
 import typer
+from rich.box import ROUNDED
+from rich.console import Console
+from rich.panel import Panel
+from rich.table import Table
+from rich.text import Text
+from rich.theme import Theme
 
-app = typer.Typer()
+from nsls2api.cli.settings import get_base_url, get_token
+from nsls2api.cli.utils.cli_helpers import auto_help_if_no_command
+
+app = typer.Typer(invoke_without_command=True)
+
+
+@app.callback()
+@auto_help_if_no_command()
+def api_callback(ctx: typer.Context):
+    pass  # No need to call anything manually
+
+
+console = Console(
+    theme=Theme(
+        {
+            "info": "cyan",
+            "warning": "yellow",
+            "error": "red bold",
+            "success": "green bold",
+            "health.good": "green bold",
+            "health.bad": "red bold",
+        }
+    )
+)
 
 
 @app.command()
 def status():
-    print("General status of NSLS-II API")
+    """Show general status of NSLS-II API"""
+    url = f"{get_base_url()}/v1/about"
+
+    with console.status("[cyan]Fetching API status...", spinner="dots"):
+        try:
+            with httpx.Client() as client:
+                token = get_token()
+                headers = {"Authorization": f"{token}"} if token else {}
+                response = client.get(url, headers=headers)
+                response.raise_for_status()
+                about_data = response.json()
```

Comment:
Because this access pattern gets reused frequently, it could be worthwhile to refactor it into into a context manager (in some future PR).  

Then it could be used in this way (or something similar):
```
with http_response_json as about_data:
```

Reply:
Indeed - and you can see I thought similar and created `call_nsls2api_endpoint()` - I just didn't do a good job and actually use it everywhere! 😄

Reply:
As the generic `call_nsls2_endpoint()` function does more that just this - I now have some error handling at multiple levels - the generic "The http call failed" and then the more specific "I didn't get the info I needed" from the 'higher' level... So, this ends up with multiple errors being displayed - does this look ok ?

<img width="1092" alt="image" src="https://github.com/user-attachments/assets/0ba0edad-e76e-4aae-9159-5800a262af39" />

Reply:
Nicely done!